ch_tools/chadmin/internal/dictionary.py

```diff
@@ -43,3 +47,333 @@ def reload_dictionary(
 
     query = f"""SYSTEM RELOAD DICTIONARY {full_name}"""
     execute_query(ctx, query, format_=None)
+
+
+def migrate_dictionaries(ctx: Context) -> None:
+    """
+    Migrate external dictionaries to DDL.
+    """
+    logging.info("Creating '_dictionaries' database")
+    execute_query(ctx, "CREATE DATABASE IF NOT EXISTS _dictionaries", format_=None)
+
+    config_glob_pattern = _get_dictionaries_config_pattern()
+    logging.debug("External dictionary pattern: {}", config_glob_pattern)
+
+    config_path = Path(CLICKHOUSE_SERVER_CONFIG_PATH)
+    config_directory = config_path.parent
+    config_path_list = list(config_directory.glob(config_glob_pattern))
+
+    logging.info("Starting external dictionaries migration")
+    for i, config_file in enumerate(config_path_list, start=1):
+        logging.info("Migration #{}: dictionary config file '{}'", i, config_file)
+        try:
+            queries = generate_ddl_dictionaries_from_xml(str(config_file))
+            for j, query in enumerate(queries, start=1):
+                logging.debug("Query #{}:\n{}", j, query)
+                execute_query(ctx, query, format_=None)
+        except Exception as error:
+            logging.exception(
+                "Failed to migrate dictionary from config file '{}'", config_file
+            )
+            logging.critical("Error message: {}", error)
+            raise RuntimeError(
+                f"Dictionary migration failed for config file '{config_file}'"
+            ) from error
+    logging.info("External dictionaries migration completed successfully")
+
+    for path in config_path_list:
+        try:
+            path.unlink()
+            logging.info("Deleted config file '{}'", path)
+        except Exception as error:
+            logging.critical("Error removing '{}': {}", path, error)
+            return
+
+
+def generate_ddl_dictionaries_from_xml(config_path: str) -> list[str]:
+    """
+    Parse XML dictionary config and generate CREATE DICTIONARY statements.
+    """
+    config: dict[str, Any] = _load_config(config_path)
+    dictionaries = config.get("dictionaries")
```

Comment:
Config is already loaded inside `_get_dictionaries_config_pattern`, it should be loaded only once and passed as an argument to other methods that need it